packages/pluggableWidgets/gallery-text-filter-native/src/components/FilterComponent.tsx

```diff
@@ -0,0 +1,112 @@
+import { createElement, ReactElement, useRef, useState, useEffect, useCallback } from "react";
+import { View, TextInput, TouchableOpacity, ViewStyle } from "react-native";
+import Svg, { Path } from "react-native-svg";
+import { debounce } from "@mendix/piw-utils-internal";
+import { extractStyles } from "@mendix/pluggable-widgets-tools";
+
+import { GalleryTextFilterStyle, InputStyleProps } from "../ui/Styles";
+
+export interface FilterComponentProps {
+    delay: number;
+    placeholder?: string;
+    styles?: GalleryTextFilterStyle;
+    updateFilters?: (value: string) => void;
+    value?: string;
+    name: string;
+}
+
+const textInputPropsKeys: Array<keyof InputStyleProps> = [
+    "autoCapitalize",
+    "placeholderTextColor",
+    "selectionColor",
+    "underlineColorAndroid"
+];
+
+export default function FilterComponent(props: FilterComponentProps): ReactElement {
+    const [value, setValue] = useState("");
+    const [valueInput, setValueInput] = useState("");
+    const [textInputContainerStyle, setTextInputContainerStyle] = useState<ViewStyle>(
+        props.styles?.textInputContainer || {}
+    );
+    const [renderClearTextIcon, setRenderClearTextIcon] = useState<boolean>(false);
+
+    const inputRef = useRef<TextInput | null>(null);
+
+    const [textInputProps, textInput] = extractStyles(props.styles?.textInput, textInputPropsKeys);
+    const xIconSVG = (
+        <Svg width={24} height={24} fill="none">
+            <Path
+                d="M6 18 18 6M18 18 6 6"
+                stroke="#606671"
+                strokeWidth={1.8}
+                strokeLinecap="round"
+                strokeLinejoin="round"
+            />
+        </Svg>
+    );
+
+    useEffect(() => setRenderClearTextIcon(valueInput !== ""), [valueInput]);
+
+    useEffect(() => {
+        props.updateFilters?.(value);
+    }, [value]);
+
+    useEffect(() => {
+        setValueInput(props.value ?? "");
+        setValue(props.value ?? "");
+    }, [props.value]);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const onChange = useCallback(
+        debounce((value: string) => setValue(value), props.delay),
+        [props.delay]
+    );
+
+    const focusInput = useCallback(() => {
+        if (inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, [inputRef]);
+
+    const onFocus = (): void =>
+        setTextInputContainerStyle({
+            ...textInputContainerStyle,
+            ...props.styles?.textInputContainerOnFocus
+        });
+
+    const onBlur = (): void => props.styles && setTextInputContainerStyle(props.styles.textInputContainer);
```

Comment:
`onChange` is wrapped with useCallback, do we really need it there? Is it expensive enough?

Reply:
It is wrapped with useCallback because of the `debounce` otherwise it will create a new instance on every render. Also, it is the same approach as the web. If you have a suggestion, I can do it that way.